src/SFA.DAS.EmployerIncentives.Functions.LegalEntities/RefreshVendorRegistrationCaseStatus.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EmployerIncentives.Functions.LegalEntities.Services.LegalEntities;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Functions.LegalEntities
+{
+    public class RefreshVendorRegistrationCaseStatus
+    {
+        private readonly IVendorRegistrationFormService _vendorRegistrationFormService;
+        private readonly IVrfCaseRefreshConfiguration _configuration;
+        private readonly IDateTimeProvider _dateTimeProvider;
+
+        public RefreshVendorRegistrationCaseStatus(IVendorRegistrationFormService vendorRegistrationFormService,
+            IVrfCaseRefreshConfiguration configuration,
+            IDateTimeProvider dateTimeProvider)
+        {
+            _vendorRegistrationFormService = vendorRegistrationFormService;
+            _configuration = configuration;
+            _dateTimeProvider = dateTimeProvider;
+        }
+
+        // Every hour: "* 0 * * * *"
+        [FunctionName("RefreshVendorRegistrationCaseStatus")]
+        public async Task Run([TimerTrigger("%RefreshVendorRegistrationCaseStatusTriggerTime%")] TimerInfo myTimer, ILogger log)
+        {
+            var from = await _configuration.GetLastRunDateTime();
```

Comment:
Would it be better to put GetLastRunDateTime, GetCurrentDateTime and UpdateLastRunDateTimein the service?  Then at least the function can be dumb as it should be and the logic is contained in the service?

Then you can decorate the service to add the logging and keep the concerns separate.

Reply:
not really